src/services/lease-service/property-rental-rules-validator.ts

```diff
@@ -1,46 +1,46 @@
-import { Tenant } from 'onecore-types'
+import { Lease } from 'onecore-types'
+import { match, P } from 'ts-pattern'
+
 import { getEstateCodeFromXpandByRentalObjectCode } from './adapters/xpand/estate-code-adapter'
 
-const propertiesWithSpecificRentalRules = [
-  {
-    estateCode: '24104', //Sjöodjuret 2
-  },
-  {
-    estateCode: '23002', //ROTORN 13
-  },
-  {
-    estateCode: '23003', //ISOLATORN 14
-  },
-]
-
-const doesPropertyBelongingToParkingSpaceHaveSpecificRentalRules = (
-  estateCode: string
-) => {
-  return propertiesWithSpecificRentalRules.some(
-    (property) => estateCode == property.estateCode
-  )
+export const PROPERTIES_WITH_SPECIFIC_RENTAL_RULES = {
+  SJOODJURET_2: '24104',
+  ROTORN_13: '23002',
+  ROTORN_14: '23001',
+  ISOLATORN_14: '23003',
 }
 
-async function doesTenantHaveHousingContractInSamePropertyAsListing(
-  data: Pick<Tenant, 'currentHousingContract' | 'upcomingHousingContract'>,
+const parkingSpaceNeedsValidation = (estateCode: string) =>
+  Object.values(PROPERTIES_WITH_SPECIFIC_RENTAL_RULES).includes(estateCode)
+
+async function isParkingSpaceRentableForTenant(
+  lease: Lease,
   listingEstateCode: string
-): Promise<boolean> {
-  const propertyInfo = data.upcomingHousingContract
-    ? await getEstateCodeFromXpandByRentalObjectCode(
-        data.upcomingHousingContract.rentalPropertyId
-      )
-    : data.currentHousingContract
-      ? await getEstateCodeFromXpandByRentalObjectCode(
-          data.currentHousingContract.rentalPropertyId
-        )
-      : undefined
+) {
+  const propertyInfo = await getEstateCodeFromXpandByRentalObjectCode(
+    lease.rentalPropertyId
+  )
 
   if (!propertyInfo) return false
 
-  return propertyInfo.estateCode === listingEstateCode
-}
+  /**
+   * ROTORN 13 and ROTORN 14 are considered the same property
+   * in the context of rental rules.
+   *
+   * So we need to check if both the listing estate code and
+   * property estate code are either ROTORN 13 or ROTORN 14.
+   * If they are, we can consider them the same property.
+   *
+   * Otherwise we expect them to be the same.
+   */
+  const { ROTORN_13, ROTORN_14 } = PROPERTIES_WITH_SPECIFIC_RENTAL_RULES
 
-export {
-  doesPropertyBelongingToParkingSpaceHaveSpecificRentalRules,
-  doesTenantHaveHousingContractInSamePropertyAsListing,
+  return match([listingEstateCode, propertyInfo.estateCode])
```

Comment:
I might be missing something here but I think some of the logic here is being obscured by the ROTORN part, at least when reading this at first glance. 

The `otherwise`-part is the default logic for all other properties, right? Is it possible to split the return into two cases to make it clear that ROTORN is a special case? I understand that ts-pattern is handling this for us now but for someone not familiar with the lib or our business logic all other real east properties are obscured by ROTORN now. I might not be right about this but it's something I reacted to. What do you think?  I also might be overthinking 😇

Reply:
Yeah, the `otherwise`-part is the default case. Do you mean that the other cases feel obscured because there is a "block" of code dedicated to the ROTOR stuff and the default case "looks" lost there?

If we had separate returns, something like this?
```ts
  const isRotor =
    (propertyInfo.estateCode === ROTORN_13 ||
      propertyInfo.estateCode === ROTORN_14) &&
    (listingEstateCode === ROTORN_13 || listingEstateCode === ROTORN_14)

  if (isRotor) {
    return true
  } else {
    return propertyInfo.estateCode === listingEstateCode
  }
```

That would be the equivalent without using `ts-pattern`.
I guess it has a lot to do with how we're used to reading code. I prefer the `ts-pattern` approach since it's so much easier to convey more complex pattern matching without making a mess. There _has been_ a special case added so it needs to be accounted for somewhere. For me it makes sense that the special cases are explicitly handled in the match function, and the default case just goes to "otherwise". 🤔

Reply:
> Do you mean that the other cases feel obscured because there is a "block" of code dedicated to the ROTOR stuff and the default case "looks" lost there?

Exactly. I guess what I don't like is that the special case gets precedence over the default case. But I can live with that and I think the code got better with this lib so I'm happy to see this refactor. 

> That would be the equivalent without using ts-pattern.

And that is not nearly as nice as using ts-pattern hehe. Some of those heavy if-statements that I wrote previously never felt good writing and were quite hard to read so I think this is a big improvement overall.

Reply:
:) Also I didn't mean that the non-ts-pattern-equivalent is "bad" or anything like that, just trying to understand what you were saying.